lib/block_view.go

```diff
@@ -4199,6 +4269,12 @@ func (bav *UtxoView) _connectSubmitPost(
 		copy(recloutedPostHash[:], recloutedPostHashBytes)
 		delete(extraData, RecloutedPostHash)
 	}
+	// Set the IsPinned attribute based on extra data and check that we've passed the PinnedPostTrackingUpdateBlockHeight.
+	isPinned := false
+	if _, hasPinned := extraData[IsPinnedPostKey]; hasPinned && blockHeight > PinnedPostTrackingUpdateBlockHeight {
+		isPinned = true
+		delete(extraData, IsPinnedPostKey)
```

Comment:
Why do we modify the extraData here?

Reply:
Similar to why we delete the reclout keys from the extra data map - it saves space in the DB for each PostEntry as we don't need the keys in the PostEntry's PostExtraData map since we explicitly extract it to the IsPinned attribute.